test/parser-test.ts

```diff
@@ -25,6 +25,40 @@ describe('Parser', () => {
     });
   });
 
+  it('Two method syntax', () => {
+    const sourceCode = `
+      /**
+      * @shouldExport true
+      */
+      interface MockedInterface {
+        mockedMethod(): void;
+        mockedFunctionProperty: () => void;
```

Comment:
Do you mean function return type? Or high-order functions? `() => void`

Reply:
Latter. In TypeScript `interface`, `mockedMethod()` is treated as `MethodSignature`, and syntax like `mockedProperty: AnyType` would be treated as `PropertySignature`, no matter the `AnyType` is an arrow function or not. So we need do have different logic to parse these two method types.

Reply:
Cool! Thanks for the explanation!